.github/workflows/deploy-orchard1-to-azure-app-service.yml

```diff
@@ -0,0 +1,201 @@
+      application-insights-resource-id:
+        required: false
+        type: string
+        default: ""
+        description: >
+          ID of the Application Insights resource that the release annotation for the deployment should be added to.
+          This can e.g. be looked up on the Azure Portal under the given AI resource's Overview page -> JSON View.
+      maintenance-host-name:
+        required: false
+        type: string
+        description: >
+          The hostname of the Orchard 1-based application that runs a maintenance based on the Multi Tenancy feature
+          (https://github.com/Lombiq/Hosting-Multi-Tenancy#writing-maintenance-providers).
+      maintenance-batch-size:
+        required: false
+        type: string
+        default: "0"
+        description: The batch size of the maintenance.
+
+jobs:
+  deploy:
+    runs-on: ${{ inputs.machine-type }}
+    name: Deploy to Azure App Service
+    defaults:
+      run:
+        shell: pwsh
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+    steps:
+      - name: Checkout
+        uses: Lombiq/GitHub-Actions/.github/actions/checkout@issue/LMBQ-105
+        with:
+          token: ${{ secrets.CHECKOUT_TOKEN }}
+
+      - name: Enable Node corepack
+        uses: Lombiq/GitHub-Actions/.github/actions/enable-corepack@issue/LMBQ-105
+
+      # Calling nuget restore separately on the actual solution, because we're passing Orchard.proj to the msbuild
+      # action instead to be able to call the Precompiled target.
+      - name: Restore NuGet packages
+        run: nuget restore ${{ inputs.build-directory }}\${{ inputs.solution-or-project-path }}
+
+      - name: Publish Precompiled App
+        uses: Lombiq/GitHub-Actions/.github/actions/msbuild@issue/LMBQ-105
+        with:
+          solution-or-project-path: Orchard.proj
+          verbosity: ${{ inputs.build-verbosity }}
+          treat-warnings-as-errors: ${{ inputs.build-treat-warnings-as-errors }}
+          enable-code-analysis: ${{ inputs.build-enable-code-analysis }}
+          msbuild-switches: |
+            /t:Precompiled
+            /p:Solution=${{ inputs.build-directory }}\${{ inputs.solution-or-project-path }}
+
+      - name: Login to Azure
+        uses: Lombiq/GitHub-Actions/.github/actions/login-to-azure@issue/LMBQ-105
+        env:
+          SERVICE_PRINCIPAL: ${{ secrets.AZURE_APP_SERVICE_DEPLOYMENT_SERVICE_PRINCIPAL }}
+
+      - name: Initialize PowerShell modules
+        uses: Lombiq/Infrastructure-Scripts/.github/actions/initialize@dev
+
+      - name: Stop Web App Slot
+        run: |
+          Stop-AzureWebAppSlot `
+            -ResourceGroupName ${{ inputs.resource-group-name }} `
+            -WebAppName ${{ inputs.app-name }} `
+            -SlotName ${{ inputs.slot-name }}
+
+      # Give the App Slot some time to actually stop, otherwise the deployment might go wrong and the app won't start,
+      # even though the deployment succeeds without warnings or errors.
```

Comment:
Is this a problem only with O1 apps? Because if not, then we should add this to the OC deploy workflow too.

Reply:
I haven't worked on the GHA-side of the OC deploys, but AFAIK this didn't come up there.